src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -0,0 +1,452 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Abstractions;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.FileSystemGlobbing.Internal;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Platform;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+    using static VirtualClient.Actions.LzbenchExecutor;
+
+    /// <summary>
+    /// The HPL(High Performance Linpack) workload executor.
+    /// </summary>
+    [UnixCompatible]
+    public class HPLinpackExecutor : VirtualClientComponent
+    {
+        private IFileSystem fileSystem;
+        private ISystemManagement systemManagement;
+        private IStateManager stateManager;
+        private IPackageManager packageManager;
+
+        /// <summary>
+        /// Constructor for <see cref="HPLinpackExecutor"/>
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public HPLinpackExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+             : base(dependencies, parameters)
+        {
+            this.systemManagement = this.Dependencies.GetService<ISystemManagement>();
+            this.packageManager = this.systemManagement.PackageManager;
+            this.fileSystem = this.systemManagement.FileSystem;
+            this.stateManager = this.systemManagement.StateManager;
+        }
+
+        /// <summary>
+        /// The order of the coefficient matrix also known as problem size (N)
+        /// </summary>
+        public string N
```

Comment:
Let's spell this out a bit more clearly. I know that Linpack may use 'N' to describe this but it would be better with a name such as ProblemSize or ProblemSizeN